### PR TITLE
Support specifying zones to create NodePools for Hypershift

### PR DESCRIPTION
**Pull request by fxierh**

### CODE REVIEW

**Comment by fxierh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-ipi-ovn-hypershift-guest-f7 periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-ovn-hypershift-guest-f7 pull-ci-openshift-hypershift-release-4.15-e2e-conformance

**Comment by fxierh** [comment]
TODO: adjust the name of the newly-introduced zone param to avoid name collision with the hypershift-aws-setup-nested-management-cluster chain.

**Comment by fxierh** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-aws-ipi-ovn-hypershift-guest-f7 periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-ovn-hypershift-guest-f7 pull-ci-openshift-hypershift-release-4.15-e2e-conformance

**Comment by fxierh** [comment]
/assign @csrwng @enxebre
/assign @heliubj18 @LiangquanLi930 

Would you please take a look ? Thanks.

File: ci-operator/step-registry/hypershift/aws/create/hypershift-aws-create-chain.yaml [hunk 1]
```diff
@@ -13,6 +13,9 @@ chain:
     - name: HYPERSHIFT_AWS_REGION
       default: "us-east-1"
       documentation: "The AWS region of the cluster."
+    - name: HYPERSHIFT_HC_ZONES
```
Comment (LiangquanLi930) [inline ci-operator/step-registry/hypershift/aws/create/hypershift-aws-create-chain.yaml:16]: Why don't we use `EXTRA_ARGS`?

Comment (fxierh) [reply ci-operator/step-registry/hypershift/aws/create/hypershift-aws-create-chain.yaml:16]: This param is added here so we don't have to pass unsupported flags through `EXTRA_ARGS`.  
Btw we already have an equivalent parameter called `HYPERSHIFT_ZONES` for `chain: hypershift-aws-setup-nested-management-cluster`.

**Pull request merged**